nlubridge/datasets.py

```diff
@@ -186,7 +190,7 @@ def sample(self, size=0.1, random_state=0, stratification="intents"):
         return sampled
 
     def filter_by_intent_name(
-        self, excluded: List[str] = None, allowed: List[str] = None
+        self, excluded: List[str] = None, allowed: List[Optional[str]] = None
```

Comment:
Verstehe ich richtigm, dass `List[Optional[str]]` bedeutet eine Liste mit strings oder None? Hier würde man doch nie None übergeben wollen, oder?

Reply:
Wahrscheinlich kommt das in der Praxis tatsächlich nicht vor. `mypy` hat aber streng genommen recht, dass im Falle einer Intent-Liste mit einzelnen `None`-Elementen, spätestens in Zeile 234 die Funktion so aufgerufen werden würde. Das finde ich eigentlich ganz cool, dass `mypy` hier erkennt, wo sich Typing-Deklarationen teilweise mit anderen Code-Stellen beißen.
Bei der Gelegenheit ist mir aber aufgefallen, dass bei `excluded` und `allowed` auch noch ein umschließendes `Optional` fehlt - seltsamerweise hat das mypy nicht bemerkt....

Reply:
Meinst du weil None in den unique intents auftauchen würde?